jump/src/context.rs

```diff
@@ -49,111 +53,195 @@ fn path_to_str(path: &Path) -> Result<&str, String> {
+pub(crate) struct Context<'a> {
+    scie: &'a Path,
+    jump: &'a Jump,
+    lift: &'a Lift,
     base: PathBuf,
-    files_by_name: BTreeMap<String, File>,
-    pub(crate) files: Vec<File>,
-    pub(crate) replacements: HashSet<File>,
-    pub(crate) description: Option<String>,
+    files_by_name: BTreeMap<&'a str, &'a File>,
+    replacements: HashSet<&'a File>,
+    bindings: IndexMap<&'a str, Binding>,
 }
 
 fn try_as_str(os_str: &OsStr) -> Option<&str> {
     <[u8]>::from_os_str(os_str).and_then(|bytes| std::str::from_utf8(bytes).ok())
 }
 
-impl Context {
+impl<'a> Context<'a> {
     #[time("debug")]
-    pub(crate) fn new(scie_path: PathBuf, lift: Lift) -> Result<Self, String> {
+    fn new(scie: &'a Path, jump: &'a Jump, lift: &'a Lift) -> Result<Self, String> {
         let mut files_by_name = BTreeMap::new();
         for file in &lift.files {
-            files_by_name.insert(file.name.clone(), file.clone());
+            files_by_name.insert(file.name.as_str(), file);
             if let Some(key) = file.key.as_ref() {
-                files_by_name.insert(key.clone(), file.clone());
+                files_by_name.insert(key.as_str(), file);
             }
         }
         Ok(Context {
-            scie: scie_path,
-            description: lift.description,
-            commands: lift.boot.commands,
-            _bindings: lift.boot.bindings,
-            base: expanduser(lift.base)?,
+            scie,
+            jump,
+            lift,
+            base: expanduser(&lift.base)?,
             files_by_name,
-            files: lift.files,
             replacements: HashSet::new(),
+            bindings: IndexMap::new(),
         })
     }
 
-    #[cfg(target_family = "windows")]
-    fn scie_basename(&self) -> Option<&str> {
-        self.scie.file_stem().and_then(try_as_str)
+    fn prepare_process(&mut self, cmd: &'a Cmd) -> Result<Process, String> {
+        let vars = cmd
+            .env
+            .iter()
+            .map(|(key, value)| {
+                self.reify_string(value)
+                    .map(|v| (EnvVar::from(key), OsString::from(v)))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        let exe = self.reify_string(&cmd.exe)?.into();
+        let args = cmd
+            .args
+            .iter()
+            .map(|string| self.reify_string(string).map(OsString::from))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        Ok(Process {
+            env: EnvVars { vars },
+            exe,
+            args,
+        })
     }
 
-    #[cfg(not(target_family = "windows"))]
-    fn scie_basename(&self) -> Option<&str> {
-        self.scie.file_name().and_then(try_as_str)
+    fn prepare(&mut self, cmd: &'a Cmd) -> Result<(Process, Vec<Binding>, Vec<FileEntry>), String> {
+        let process = self.prepare_process(cmd)?;
+
+        let mut file_entries = vec![];
+        for file in &self.lift.files {
+            if self.replacements.contains(&file) {
```

Comment:
This is the bug introduced. If there is a replacement for, say `{node}`, that is added as `FileEntry::Install` - so far so good. If that file thogh has zero size (which indicates its stored in the scie-tote), the scie-tote needs to be added as a `FileEntry::Install` as well.